pkg/services/folder/folderimpl/folder.go

```diff
@@ -635,6 +635,19 @@ func (s *Service) Delete(ctx context.Context, cmd *folder.DeleteFolderCommand) e
 				if err := s.deleteChildrenInFolder(ctx, dashFolder.OrgID, dashFolder.UID, cmd.SignedInUser); err != nil {
 					return err
 				}
+			} else {
+				alertRuleSrv, ok := s.registry[entity.StandardKindAlertRule]
+				if !ok {
+					return folder.ErrInternal.Errorf("no alert rule service found in registry")
+				}
+				alertRulesInFolder, err := alertRuleSrv.CountInFolder(ctx, dashFolder.OrgID, dashFolder.UID, cmd.SignedInUser)
+				if err != nil {
+					s.log.Error("failed to count alert rules in folder", "error", err)
+					return err
+				}
+				if alertRulesInFolder > 0 {
+					return folder.ErrFolderContainsAlertRules.Errorf("")
```

Comment:
Should `Errorf("")` have a message here?

Reply:
Not necessary because Errorf argument changes the private message and log messages. Whereas the public message gets returned to the user here https://github.com/grafana/grafana/blob/e4d1fdc3d00fdb7dc22227aea17186a01000cb2c/pkg/api/response/response.go#L272-L274 and in a few other places.

Anyway, I think we can return just `folder.ErrFolderContainsAlertRules` because `errutil.Base` implements the interface.

As a side note, I think we should add method `Error` to the `errutil.Base` so we can get just a plain simple `errutil.Error` and avoid controversial `Errorf("")`.